### DISCUSSION TITLE
Use proxy with ConnectTimeout

### DISCUSSION THREAD

**Discussion by Mystic8b** [eternnoir/pyTelegramBotAPI#2328] [Q&A]
Greetings, friends.
I couldn't find a good solution for using a proxy when the machine is offline.
I tried doing this:
```
    except ConnectTimeout as c:
        apihelper.proxy = {'http': envsettings.tg_proxy, 'https': envsettings.tg_proxy}
        _send(text=message, effect=effect_id)
```
But in this case, the proxy will be used for all subsequent messages.
If I add `apihelper.proxy = {}` anywhere, it causes a multithreading problem (in the main program code). While one thread disables the proxy, another is still relying on it, and some notifications are not sent.
I thought about it all day yesterday and couldn't find a nice solution. Of course, I can check the internet availability on the machine in a separate process, but it doesn't seem like the best solution to me.

**Comment by coder2020official** [comment]
I don't think we can do anything about it. Proxies are usually used fully or not used at all
